src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Here is a very long callback, let's divide this.
We should to create a new helper function that check if the child has a mother. 
After that we can decrease our code and make our code more readable
```suggestion
  const filteredByChildren = people.filter((person) => onlyWithSon
    ? person.sex === 'm' && checkIsHasMother(person.mother, people)
    : checkIsHasMother(person.mother, people)
  );
  ```